addons/stock/models/stock_picking.py

```diff
@@ -910,9 +910,12 @@ def _set_scheduled_date(self):
             picking.move_ids.write({'date': picking.scheduled_date})
 
     def _has_scrap_move(self):
+        result = dict(self.env['stock.move']._read_group(
+            [('picking_id', 'in', self.ids), ('scrapped', '=', True)],
+            ['picking_id'], ['__count']
+        ))
```

Comment:
You actually don't need to count anything.
```suggestion
        result = {
             for [picking] in 
             self.env['stock.move']._read_group([('picking_id', 'in', self.ids), ('scrapped', '=', True)], ['picking_id'])
        }
```

(of course you need to change your condition)